### DISCUSSION TITLE
Container prefix instead of full name

### DISCUSSION THREAD

**Discussion by exander77** [nginx-proxy/acme-companion#952] [Ideas]
Some services like k8s on TrueNAS generate container names dynamically using a prefix and some auto-generated ids, I suggest to add something like this to accommodate:

```diff
diff --git a/app/functions.sh b/app/functions.sh
index bec4664..331e7de 100644
--- a/app/functions.sh
+++ b/app/functions.sh
@@ -290,6 +290,8 @@ function get_nginx_proxy_container {
         # ... and the env var is set, use it ...
         if [[ -n "${NGINX_PROXY_CONTAINER:-}" ]]; then
             nginx_cid="$NGINX_PROXY_CONTAINER"
+        elif [[ -n "${NGINX_PROXY_CONTAINER_PREFIX:-}" ]]; then
+            nginx_cid=$(docker_api "/containers/json" | jq .[].Names 2>/dev/null | jq -r .[] 2>/dev/null | sed "s/^\///" | grep "$NGINX_PROXY_CONTAINER_PREFIX")
         # ... else try to get the container ID with the volumes_from method.
         elif [[ $(get_self_cid) ]]; then
             volumes_from=$(docker_api "/containers/$(get_self_cid)/json" | jq -r '.HostConfig.VolumesFrom[]' 2>/dev/null)
```

**Comment by buchdag** [comment]
Hi.

I think it's already handled by the [`label` method](https://github.com/nginx-proxy/acme-companion/blob/main/docs/Getting-containers-IDs.md):

> The advantage the `label` methods have over the `environment variable` (and `volumes_from`) methods is enabling the use of the **acme-companion** in environments where containers names are dynamic, like in Swarm Mode or in Docker Cloud. Howhever if you intend to do so, as upstream **docker-gen** lacks the ability to identify containers from labels, you'll need both to either use the two containers setup or to replace nginx-proxy/docker-gen with a fork that has this ability like [herlderco/docker-gen](https://github.com/helderco/docker-gen). Be advised that for now, this works to a very limited extent [(everything has to be on the same node)](https://github.com/nginx-proxy/acme-companion/pull/231#issuecomment-330624331).
